ddtrace/propagation/http.py

```diff
@@ -1129,27 +1134,45 @@ def my_controller(url, headers):
         :param dict headers: HTTP headers to extract tracing attributes.
         :return: New `Context` with propagated attributes.
         """
-        if not headers:
+        if not headers or config._propagation_behavior_extract == _PROPAGATION_BEHAVIOR_IGNORE:
             return Context()
         try:
-            normalized_headers = {name.lower(): v for name, v in headers.items()}
+            normalized_headers: Dict = {name.lower(): v for name, v in headers.items()}
             context = Context()
-            # tracer configured to extract first only
             if config._propagation_extract_first:
-                # loop through the extract propagation styles specified in order, return whatever context we get first
+                # use only first propagation style specified, even if it doesn't find a context
+                # we don't try the other styles
                 for prop_style in config._propagation_style_extract:
                     propagator = _PROP_STYLES[prop_style]
-                    context = propagator._extract(normalized_headers)
+                    rec_context = propagator._extract(normalized_headers)
+                    # if configured to restart, append our first context to the current context as a span link
+                    if rec_context:
```

Comment:
Slightly changed behavior here so that the context never is set to `None`. If this goes against spec I can change back.